crates/wasi-component-loader/src/extension/runtime/authentication.rs

```diff
@@ -65,4 +65,20 @@ impl AuthenticationExtension<SharedContext> for WasmExtensions {
 
         (Arc::into_inner(headers).unwrap(), result)
     }
+
+    async fn public_metadata(
+        &self,
+        extension_ids: &[ExtensionId],
+    ) -> Result<Vec<runtime::authentication::PublicMetadataEndpoint>, String> {
+        let mut endpoints = Vec::new();
+
+        for id in extension_ids {
+            let mut instance = self.get(*id).await.unwrap();
+
+            let new_endpoints = instance.public_metadata().await.map_err(|err| err.to_string())?;
+            endpoints.extend(new_endpoints.into_iter());
```

Comment:
nitpick, think `append` or `extend_from_slice` would be faster

Reply:
The iterator should be sized and extend should take advantage of that, but I have to admit I just don't know.